### PR TITLE
[GH-52] Use Jackson for JSON handling

### LINKED ISSUE #52: Remove extra JSON libs
### Subject of the issue

Kafka includes Jackson. There is no need to use Gson **and** `org.json` libs. 

### Steps to reproduce
`gradle build`
### Expected behaviour
`gradle :dependencies` should not show extra JSON libs used in implementation
### Actual behaviour
It does

### PR DESCRIPTION
**Pull request by OneCricketeer**
## Summary

- Replaces Gson and org.json functions with Jackson
- Add Connect Schema support via built-in JSONConverter class

## Related Issue

Closes #52 

## How Has This Been Tested?

`gradle test`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: streaming-connect-sink/src/test/java/com/adobe/platform/streaming/sink/utils/SinkUtilsTest.java [hunk 1]
```diff
@@ -59,5 +77,65 @@ void testGetPropertyWithMultiplier() {
...
+    void testPrimitiveValueSchemaAsJson(Schema valueSchema, Object value, String expectedJson) {
+      SinkRecord sr = getSinkRecordWithValue(valueSchema, value);
+      assertDoesNotThrow(() -> assertEquals(expectedJson, SinkUtils.getStringPayload(jsonConverter, sr)));
+    }
+
+    static Stream<Object[]> primitiveSchemaPayloads() throws JsonProcessingException {
+      byte[] bytes = "secret".getBytes(StandardCharsets.UTF_8);
+      return Stream.of(
+          new Object[]{Schema.STRING_SCHEMA, "test", "test"},
+          // returns null string
```
Comment (OneCricketeer) [inline streaming-connect-sink/src/test/java/com/adobe/platform/streaming/sink/utils/SinkUtilsTest.java:107]:
```suggestion
          // does not return "null" as string
```

File: streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/impl/AEPPublisher.java [hunk 2]
```diff
@@ -18,17 +18,22 @@
 import com.adobe.platform.streaming.http.HttpProducer;
 import com.adobe.platform.streaming.http.HttpUtil;
 import com.adobe.platform.streaming.sink.AbstractAEPPublisher;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
```
Comment (vivetiwa) [inline streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/impl/AEPPublisher.java:23]: Unused imports. Please remove

File: streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/impl/AEPPublisher.java [hunk 3]
```diff
@@ -18,17 +18,22 @@
...
 import org.apache.commons.lang3.tuple.Pair;
-import org.apache.http.entity.ContentType;
 import org.apache.kafka.connect.sink.ErrantRecordReporter;
 import org.apache.kafka.connect.sink.SinkRecord;
-import org.json.JSONArray;
-import org.json.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.nio.charset.StandardCharsets;
```
Comment (vivetiwa) [inline streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/impl/AEPPublisher.java:35]: Unused imports.

**Review by vivetiwa — APPROVED** [review]
LGTM! thanks for the changes.

File: streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/utils/SinkUtils.java [hunk 4]
```diff
@@ -29,15 +29,8 @@ public class SinkUtils {
...
-      return null;
-    }
-    if (valueSchema == null) {
-      if (value instanceof String) {
-        return (String) value;
-      }
-    } else if (Schema.STRING_SCHEMA.type().equals(valueSchema.type())) {
-      return (String) value;
+    if (record.value() instanceof String) {
+      return (String) record.value();
```
[the remark this answers is not in the document]
Comment (vivetiwa) [reply streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/utils/SinkUtils.java:33]: yes, better to use `value` here. Will take care in next PR.

**Pull request merged**